app/src/main/kotlin/at/bitfire/davdroid/ui/widget/RadioWithSwitch.kt

```diff
@@ -0,0 +1,178 @@
+ *
+ * @param title The "proper" text of the Radio button. Shown in the middle of the row, between the
+ * radio button and the switch.
+ * @param summary If not `null`, shown below the title. Used to give more context or information.
+ * Supports formatting and interactions.
+ * @param isSelected Whether the item is currently selected. Refers to the radio button.
+ * @param isToggled Whether the switch is toggled.
+ * @param modifier Any modifiers to apply to the row.
+ * @param enabled Whether the radio button should be enabled. The enabled state of the switch is
+ * reverse from this. So if it's `true`, the switch will be disabled.
+ * @param onSelected Gets called whenever the user requests this row to be enabled. Either by
+ * selecting the radio button or tapping the text.
+ * @param onToggled Gets called whenever the switch gets updated. Contains the checked status.
+ */
+@Composable
+@ExperimentalTextApi
+fun RadioWithSwitch(
+    title: String,
+    summary: AnnotatedString?,
+    isSelected: Boolean,
+    isToggled: Boolean,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    onSelected: () -> Unit,
+    onToggled: (Boolean) -> Unit
+) {
+    val uriHandler = LocalUriHandler.current
+
+    Row(modifier) {
+        RadioButton(selected = isSelected, onClick = onSelected, enabled = enabled)
+
+        Column(
+            modifier = Modifier
+                .weight(1f)
+                .clickable(enabled = enabled, role = Role.RadioButton, onClick = onSelected)
+        ) {
+            Text(
+                text = title,
+                color = LocalContentColor.current.copy(
+                    alpha = if (enabled) 1f else ContentAlpha.disabled
+                ),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier.fillMaxWidth()
+            )
+            summary?.let { sum ->
+                ClickableText(
+                    text = sum,
+                    style = MaterialTheme.typography.body2.copy(
+                        color = LocalContentColor.current.copy(
+                            alpha = if (enabled) 1f else ContentAlpha.disabled
+                        )
+                    ),
+                    modifier = Modifier.fillMaxWidth(),
+                    onClick = { index ->
+                        // Get the tapped position, and check if there's any link
+                        val annotation = sum.getUrlAnnotations(index, index).firstOrNull()
+                        try {
+                            // If there is, open it
+                            annotation?.item?.url?.let(uriHandler::openUri)
+                        } catch (_: ActivityNotFoundException) {
+                            // There isn't any application available to launch the link
+                            Log.e("RadioWithSwitch", "No app available to launch ${annotation?.item?.url}")
+                        }
+                    }
+                )
+            }
```

Comment:
Can we get the annotations / formatting thing, together with the ExperimentalText annotation, out of the `RadioWithSwitch`?

Maybe we can pass the summary as Composable block, and then `ProvideTextStyle(body1) { summary() }`?